docs/docs/user/subscribing.md

```diff
@@ -152,6 +173,71 @@ is growing. First things first, you should check subscription metrics for signs
 
 If you want to know the exact algorithm, check [rate limiting configuration page](/configuration/rate-limiting/).
 
+## Additional headers
+
+Each subscription can define a number of additional `headers` that will be added to every HTTP request when sending messages.
+They can be useful on test environments to pass security tokens or on production to communicate with some legacy systems that 
+require custom headers.
+
+## Endpoint address resolver metadata
+
+Custom implementation of Consumer's `EndpointAddressResolver` interface can make use of provided `endpointAddressResolverMetadata` 
+for selecting address resolving strategies. This object deserializes to `Map<String, Object>` and may contain any data needed, 
```

Comment:
"This object deserializes to". What is deserialized? If endpointAddressResolverMetadata then maybe "This object is deserialized to ..." will be more clear.  Beside that, LGTM